tests/lib/e2e/buckets/buckets.e2e-spec.ts

```diff
@@ -0,0 +1,192 @@
+          expect(dbBucket.toObject().pubkeys).toEqual([])
+
+        })
+        it('Whe you want to update a bucket, it should fail with invalid pubkeys list', async () => {
+          // Arrange
+          const { body: bucket } = await testServer
+            .post('/buckets')
+            .set('Authorization', getAuth(testUser))
+            .send({
+              pubkeys: ['031a259ee122414f57a63bbd6887ee17960e9106b0adcf89a298cdad2108adf4d9'],
+              name: dataGenerator.word({ length: 7 })
+            })
+            .expect(201);
+
+          // Act
+          const response = await testServer
+            .patch(`/buckets/${bucket.id}`)
+            .set('Authorization', getAuth(testUser))
+            .send({ pubkeys: ['invalid-pubkey'] })
+
+
+          // Assert
+          expect(response.status).toBe(400);
+
+          const dbBucket = await engine.storage.models.Bucket.findOne({ _id: bucket.id })
+          expect(dbBucket.toObject().pubkeys).toEqual(['031a259ee122414f57a63bbd6887ee17960e9106b0adcf89a298cdad2108adf4d9'])
+
+        })
+      })
+
+      describe('Deleting a bucket', () => {
+        it('When you want to delete a bucket it should work if is the owner', async () => {
+
+          // Arrange: Create a bucket
+          const { body: bucket } = await testServer
+            .post('/buckets')
+            .set('Authorization', getAuth(testUser))
+            .send({
+              pubkeys: ['031a259ee122414f57a63bbd6887ee17960e9106b0adcf89a298cdad2108adf4d9'],
+              name: dataGenerator.word({ length: 7 })
+            })
+            .expect(201);
+
+          // Act: Delete the bucket
+          const response = await testServer
+            .delete(`/buckets/${bucket.id}`)
+            .set('Authorization', getAuth(testUser))
+
+          // Assert
+          expect(response.status).toBe(204)
+          const buckets = await engine.storage.models.Bucket.findOne({ _id: bucket.id })
+          expect(buckets).toBeNull()
+
+        })
+
+        it('When you want to delete a bucket it should fail if is not the owner', async () => {
+
+          // Arrange: Create a bucket
+          const owner = await createTestUser({ user: { email: dataGenerator.email(), password: dataGenerator.hash({ length: 64 }), maxSpaceBytes: 321312313 } })
+
+          const { body: bucket } = await testServer
+            .post('/buckets')
+            .set('Authorization', getAuth(owner))
+            .send({
+              pubkeys: ['031a259ee122414f57a63bbd6887ee17960e9106b0adcf89a298cdad2108adf4d9'],
+              name: dataGenerator.word({ length: 7 })
+            })
+            .expect(201);
+
+          // Act: Delete the bucket
+          const response = await testServer
+            .delete(`/buckets/${bucket.id}`)
+            .set('Authorization', getAuth(testUser))
+
+
+          // Assert
+          expect(response.status).toBe(404)
+
+          await deleteTestUser({ user: owner })
```

Comment:
Why is this required? Tests ideally should be as idempotent as possible. If you need to perform post cleanups, use `afterAll` hook

Reply:
This is not necessary, global-teardown cleans up the test database after all tests. I'm removing it.